precompiles/xcm/src/lib.rs

```diff
@@ -163,4 +165,79 @@ where
 
         Ok(succeed(EvmDataWriter::new().write(true).build()))
     }
+
+    fn remote_transact(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+        let mut input = handle.read_input()?;
+        input.expect_arguments(6)?;
+
+        // Raw call arguments
+        let para_id: u32 = input.read::<U256>()?.low_u32();
+        let is_relay = input.read::<bool>()?;
+
+        let fee_asset_addr = input.read::<Address>()?;
+        let fee_amount = input.read::<U256>()?;
+
+        let remote_call: Vec<u8> = input.read::<Bytes>()?.into();
+        let transact_weight = input.read::<u64>()?;
+
+        log::trace!(target: "xcm-precompile:remote_transact", "Raw arguments: para_id: {}, is_relay: {}, fee_asset_addr: {:?}, \
+         fee_amount: {:?}, remote_call: {:?}, transact_weight: {}",
+        para_id, is_relay, fee_asset_addr, fee_amount, remote_call, transact_weight);
+
+        // Process arguments
```

Comment:
The comment actually says nothing. Probably OK to remove or rephrase if that's important.

Reply:
I don't like these big methods so I put it here since I'm past raw arguments now. Guess it's a personal preference.